### PR TITLE
[FIX] account: env has no company_id but env.user has

### PR DESCRIPTION
**Pull request by wtaferner**
**Description of the issue/feature this PR addresses:**
Tax Adjustment wizard is not working

**Current behavior before PR:**
Traceback

Desired behavior after PR is merged:
No Traceback

@qdp-odoo @sswapnesh 
I am on a bug spree ;-) this one is clear at least...

Info: @wt-io-it

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/wizard/wizard_tax_adjustments.py [hunk 1]
```diff
@@ -19,8 +19,8 @@ def _get_default_journal(self):
     amount = fields.Monetary(currency_field='company_currency_id', required=True)
     adjustment_type = fields.Selection([('debit', 'Applied on debit journal item'), ('credit', 'Applied on credit journal item')], string="Adjustment Type", required=True)
     tax_report_line_id = fields.Many2one(string="Report Line", comodel_name='account.tax.report.line', required=True, help="The report line to make an adjustment for.")
-    company_currency_id = fields.Many2one('res.currency', readonly=True, default=lambda x: x.env.company_id.currency_id)
-    country_id = fields.Many2one(string="Country", comodel_name='res.country', readonly=True, default=lambda x: x.env.company_id.country_id)
+    company_currency_id = fields.Many2one('res.currency', readonly=True, default=lambda x: x.env.user.company_id.currency_id)
```
Comment (sswapnesh) [inline addons/account/wizard/wizard_tax_adjustments.py]:
```suggestion
    company_currency_id = fields.Many2one('res.currency', readonly=True, default=lambda x: x.env.company.currency_id)
```

File: addons/account/wizard/wizard_tax_adjustments.py [hunk 2]
```diff
@@ -19,8 +19,8 @@ def _get_default_journal(self):
...
+    country_id = fields.Many2one(string="Country", comodel_name='res.country', readonly=True, default=lambda x: x.env.user.company_id.country_id)
```
Comment (sswapnesh) [inline addons/account/wizard/wizard_tax_adjustments.py]:
```suggestion
    country_id = fields.Many2one(string="Country", comodel_name='res.country', readonly=True, default=lambda x: x.env.company.country_id)
```

**Comment by sswapnesh** [comment]
@wtaferner  So it seems `x.env.user.company_id` = `x.env.company`
https://github.com/odoo/odoo/blob/master/addons/web/models/ir_http.py#L55

-----------------------------
I see there is no need of `company_id` in env since https://github.com/odoo/odoo/commit/f5dfe4727c086c7577da2aef12c2d3548ba938a3#diff-2c418cb78b7c562a79e7032645a537d2L858

https://github.com/odoo/odoo/blob/master/addons/web/models/ir_http.py#L55

> "company_id": user.company_id.id if request.session.uid else None,  # YTI TODO: Remove this from the user context

@tivisse  Can we remove this ?

**Comment by wtaferner** [comment]
@sswapnesh 
Good input. I changed it to the probably intended env.company which has an additional security check if the user is allowed for the chosen company with fallback to the real user company.

I found another traceback in the code in SMS.

**Comment by sswapnesh** [comment]
I think you will need to Create Individual PR for Each module.

**Comment by tivisse** [comment]
cc: This would probably be interesting for you @tde-banana-odoo ;)

**Comment by sswapnesh** [comment]
@wtaferner  First of this PR (In Tax Adj Wizard) has been fixed in https://github.com/odoo/odoo/pull/35554

**Comment by qdp-odoo** [comment]
@wtaferner So this PR now only fix event_sms, can you change the commit message accordingly?

thanks

**Comment by wtaferner** [comment]
@qdp-odoo 
Sure, if we get it merged then, it was worth it ;-)

**Comment by qdp-odoo** [comment]
@robodoo r+

thanks Wolfgang!

**Pull request closed without merging**